api/v1alpha4/azurecluster_default_test.go

```diff
@@ -900,8 +1352,51 @@ func TestNodeOutboundLBDefaults(t *testing.T) {
 								},
 							},
 							Type:                 Public,
-							FrontendIPsCount:     to.Int32Ptr(2),
-							IdleTimeoutInMinutes: to.Int32Ptr(15),
+							FrontendIPsCount:     to.Int32Ptr(2),  // we expect the original value to be respected here
+							IdleTimeoutInMinutes: to.Int32Ptr(15), // we expect the original value to be respected here
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "NodeOutboundLB declared as input with non-respected values",
```

Comment:
@CecileRobertMichon this is a weird case to test from a UX perspective but in fact it expresses literally what happens if you pass in a predefined LB object w/ either `Name`, `SKU`, or `Type` values. So if we _want_ to statically override those values we want this UT; or, if we don't want to do that now's the time to think about what we do want to do.

Reply:
The expected behavior, documented in https://capz.sigs.k8s.io/topics/node-outbound-lb.html#warning, is that these values are not configurable and modifying them should result in a validation error. It should _not_ silently override the values. If that's what is happening right now then it's a bug. I think we should leave these values as is if already set and let the validation webhook do it's job by failing if the values are not what we expect (validation should already be in place). This also applies to the control plane outbound LB.

This is a bit out of scope for this PR so feel free to open an issue if you don't want to address this as part of this PR.

Reply:
I think what we're observing is that this condition (user-configurable value overrides of properties that aren't user-overridable) is artificial; as you say, validation will prevent this condition from ever reaching this point in the code flow. I vote that we just not add this artificial test case (I'll delete it).